aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/SyncProcessorTest.java

```diff
@@ -116,6 +130,88 @@ public void setup() throws AmplifyException {
+     * @throws DataStoreException Not expected.
+     */
+    @Test
+    public void dataStoreHubEventsTriggered() throws DataStoreException {
+        // Arrange - BEGIN
+        int expectedModelCount = Arrays.asList(Post.class, BlogOwner.class).size();
+        // Collects one syncQueriesStarted event.
+        HubAccumulator syncStartAccumulator =
+            createAccumulator(syncQueryStartedForModels(modelCount), 1);
+        // Collects one syncQueriesReady event.
+        HubAccumulator syncQueryReadyAccumulator =
+            createAccumulator(forEvent(DataStoreChannelEventName.SYNC_QUERIES_READY), 1);
+        // Collects one modelSynced event for each model.
+        HubAccumulator modelSyncedAccumulator =
+            createAccumulator(forEvent(DataStoreChannelEventName.MODEL_SYNCED), expectedModelCount);
+
+        // Add a couple of seed records so they can be deleted/updated.
+        storageAdapter.save(DRUM_POST.getModel());
+        storageAdapter.save(BLOGGER_ISLA.getModel());
+
+        // Mock sync query results for a couple of models.
+        AppSyncMocking.sync(appSync)
+            .mockSuccessResponse(Post.class, DELETED_DRUM_POST)
+            .mockSuccessResponse(BlogOwner.class, BLOGGER_ISLA, BLOGGER_JAMESON);
+
+        // Start the accumulators.
+        syncQueryReadyAccumulator.start();
+        syncStartAccumulator.start();
+        modelSyncedAccumulator.start();
+
+        TestObserver<ModelWithMetadata<? extends Model>> hydrationObserver = TestObserver.create();
+        // Arrange - END
+
+        // Act: kickoff sync.
+        syncProcessor.hydrate().subscribe(hydrationObserver);
+
+        // Check - BEGIN
+        // Verify that sync completes.
+        assertTrue(hydrationObserver.awaitTerminalEvent(OP_TIMEOUT_MS, TimeUnit.MILLISECONDS));
+        hydrationObserver.assertNoErrors();
+        hydrationObserver.assertComplete();
+
+        // Verify that syncQueriesStarted was emitted once.
+        assertEquals(1, syncStartAccumulator.await((int) OP_TIMEOUT_MS, TimeUnit.MILLISECONDS).size());
+        // Verify that syncQueriesReady was emitted once.
+        assertEquals(1, syncQueryReadyAccumulator.await((int) OP_TIMEOUT_MS, TimeUnit.MILLISECONDS).size());
+
+        // Get the list of modelSynced events captured.
+        List<HubEvent<?>> hubEvents = modelSyncedAccumulator.await((int) OP_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        // Verify that [number of events] = [number of models]
+        assertEquals(expectedModelCount, hubEvents.size());
+
+        // For each event (excluding system models), verify the desired count.
+        for (HubEvent<?> event : hubEvents) {
+            ModelSyncedEvent eventData = (ModelSyncedEvent) event.getData();
+            assertTrue(eventData.isFullSync());
+            assertFalse(eventData.isDeltaSync());
+            String eventModel = eventData.getModel();
+            switch (eventModel) {
+                case "BlogOwner":
```

Comment:
Can we do `BlogOwner.class.getSimpleName()` for these case statements, instead of string literal?

Reply:
Interesting...Evidently `BlogOwner.class.getSimpleName()` is not considered a compile-time constant. Even if I create a final static field and assign it the value of `BlogOwner.class.getSimpleName()`, the compiler still flags it with a `Constant expression required` error